### PR TITLE
Update AWS & HTTPComponents

### PR DESCRIPTION
**Pull request by KenFigueiredo**
Short PR to update some dependencies to their latest versions
* Update aws sdk version to `1.11.728`.
* Update apache httpcomponents version to `4.5.11`.
* Also updates license date on specific files
* Cherry-pick CI changes from Java 8 PR.

### CODE REVIEW

**Comment by KenFigueiredo** [comment]
It looks like the updating of the parent pom broke builds for Java 7: https://github.com/HotelsDotCom/circus-train/pull/168/files

Current error - but if I revert the parent pom to `4.2.0` the build runs as expected
```[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:1.27.0:apply (default) on project circus-train-parent: Execution default of goal com.diffplug.spotless:spotless-maven-plugin:1.27.0:apply failed: Unable to load the mojo 'apply' in the plugin 'com.diffplug.spotless:spotless-maven-plugin:1.27.0' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: com/diffplug/spotless/maven/SpotlessApplyMojo : Unsupported major.minor version 52.0
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>com.diffplug.spotless:spotless-maven-plugin:1.27.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/kfigueiredo/.m2/repository/com/diffplug/spotless/spotless-maven-plugin/1.27.0/spotless-maven-plugin-1.27.0.jar
[ERROR] urls[1] = file:/Users/kfigueiredo/.m2/repository/com/diffplug/spotless/spotless-lib/1.27.0/spotless-lib-1.27.0.jar
[ERROR] urls[2] = file:/Users/kfigueiredo/.m2/repository/com/diffplug/spotless/spotless-lib-extra/1.27.0/spotless-lib-extra-1.27.0.jar
[ERROR] urls[3] = file:/Users/kfigueiredo/.m2/repository/org/eclipse/jgit/org.eclipse.jgit/5.5.0.201909110433-r/org.eclipse.jgit-5.5.0.201909110433-r.jar
[ERROR] urls[4] = file:/Users/kfigueiredo/.m2/repository/com/jcraft/jsch/0.1.55/jsch-0.1.55.jar
[ERROR] urls[5] = file:/Users/kfigueiredo/.m2/repository/com/jcraft/jzlib/1.1.1/jzlib-1.1.1.jar
[ERROR] urls[6] = file:/Users/kfigueiredo/.m2/repository/com/googlecode/javaewah/JavaEWAH/1.1.6/JavaEWAH-1.1.6.jar
[ERROR] urls[7] = file:/Users/kfigueiredo/.m2/repository/org/bouncycastle/bcpg-jdk15on/1.61/bcpg-jdk15on-1.61.jar
[ERROR] urls[8] = file:/Users/kfigueiredo/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.61/bcprov-jdk15on-1.61.jar
[ERROR] urls[9] = file:/Users/kfigueiredo/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.61/bcpkix-jdk15on-1.61.jar
[ERROR] urls[10] = file:/Users/kfigueiredo/.m2/repository/com/googlecode/concurrent-trees/concurrent-trees/2.6.1/concurrent-trees-2.6.1.jar
[ERROR] urls[11] = file:/Users/kfigueiredo/.m2/repository/org/codehaus/groovy/groovy-xml/2.4.7/groovy-xml-2.4.7.jar
[ERROR] urls[12] = file:/Users/kfigueiredo/.m2/repository/org/codehaus/groovy/groovy/2.4.7/groovy-2.4.7.jar
[ERROR] urls[13] = file:/Users/kfigueiredo/.m2/repository/org/codehaus/plexus/plexus-resources/1.0.1/plexus-resources-1.0.1.jar
[ERROR] urls[14] = file:/Users/kfigueiredo/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.8/plexus-utils-3.0.8.jar
[ERROR] urls[15] = file:/Users/kfigueiredo/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[16] = file:/Users/kfigueiredo/.m2/repository/com/diffplug/durian/durian-core/1.2.0/durian-core-1.2.0.jar
[ERROR] urls[17] = file:/Users/kfigueiredo/.m2/repository/com/diffplug/durian/durian-collect/1.2.0/durian-collect-1.2.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>com.hotels:circus-train-parent:15.1.2-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR] 
[ERROR] -----------------------------------------------------
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [15.1.2] - 2020-02-24
```
Comment (massdosage) [inline CHANGELOG.md]: Leave the date as TBD, we'll set it when we do a release. We normally do minor version releases if we update libraries so go with 15.2.0 as the version number.

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
   directories:
     - ~/.m2/repository
 install:
-  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true --quiet --batch-mode --show-version --activate-profiles travis
+  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true --quiet --batch-mode --show-version
 before_script:
   - echo "MAVEN_OPTS=${TEST_MEM_SETTINGS}" > ~/.mavenrc
 script:
-  - mvn cobertura:cobertura coveralls:report --batch-mode --activate-profiles travis -DrepoToken=${COVERALLS_REPO_TOKEN} -Dtest.arguments="${TEST_MEM_SETTINGS}"
+  - mvn package jacoco:report coveralls:report --batch-mode -Dtest.arguments="${TEST_MEM_SETTINGS}" -P coveralls --quiet
```
Comment (massdosage) [inline .travis.yml]: Please use the long name for the arguments as they are more readable (i.e. `--activate-profiles` instead of `-P`). 

The coveralls repo token argument needs to be put back.

Why did you add `--quiet`?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: build
```
Comment (massdosage) [inline .github/workflows/main.yml]: Let's not add this file in this PR, I need to generate an add the coveralls token for the action so I'd rather do that in the other PR.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+## [15.2.0] - TBD
+### Changed
+* Updated `aws-java-sdk-s3` version to 1.11.728 (was 1.11.505).
```
Comment (massdosage) [inline CHANGELOG.md]: This is OK, you could also just have one line mentioning the property that you changed as it affects all 3 (and maybe some other dependencies in child modules).

File: README.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
...
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.hotels/circus-train/badge.svg?subject=com.hotels:circus-train)](https://maven-badges.herokuapp.com/maven-central/com.hotels/circus-train) [![Build Status](https://travis-ci.org/HotelsDotCom/circus-train.svg?branch=master)](https://travis-ci.org/HotelsDotCom/circus-train) ![build](https://github.com/HotelsDotCom/circus-train/workflows/Java%20CI/badge.svg?event=push) [![Coverage Status](https://coveralls.io/repos/github/HotelsDotCom/circus-train/badge.svg?branch=master)](https://coveralls.io/github/HotelsDotCom/circus-train?branch=master) ![GitHub license](https://img.shields.io/github/license/HotelsDotCom/circus-train.svg)
```
Comment (massdosage) [inline README.md]: Again, let's keep the GH action in the other PR.

File: pom.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>com.hotels</groupId>
     <artifactId>hotels-oss-parent</artifactId>
-    <version>4.3.1</version>
+    <version>5.0.0</version>
```
Comment (massdosage) [inline pom.xml]: Please add this change to the changelog too as changing the parent can have side effects that we sometimes need to track down later.

**Comment by massdosage** [comment]
@KenFigueiredo I merged the other PR in as I got 2 approvals today. So I think if you resolve the merge conflicts here you'll be good to go.

File: .travis.yml (hunk 2 above)
Comment (KenFigueiredo) [reply .travis.yml]: good catch, didn't realize I removed the token.
Added quiet because travis was failing with : `The job exceeded the maximum log length, and has been terminated.`, it doesn't look like it helped much. Will go ahead and revert this

Comment (KenFigueiredo) [reply .travis.yml]: ah right - I had copied that directly from the other PR. Do we need the token call still?

File: .travis.yml [hunk 7]
```diff
@@ -11,4 +11,4 @@ install:
 before_script:
   - echo "MAVEN_OPTS=${TEST_MEM_SETTINGS}" > ~/.mavenrc
 script:
-  - mvn jacoco:report coveralls:report --batch-mode -Dtest.arguments="${TEST_MEM_SETTINGS}"
+  - mvn package jacoco:report coveralls:report --batch-mode --activate-profiles coveralls -Dtest.arguments="${TEST_MEM_SETTINGS}"
```
Comment (massdosage) [inline .travis.yml]: Sorry, partly my mistake in my merge, I intended to move coveralls over to the GH action and not in Travis so we don't get two reports. Can you please remove `coveralls:report` and `--activate-profiles coveralls` from here. `package` and `jacoco` are enough.
